mlflow/tracking/client.py

```diff
@@ -496,6 +498,24 @@ def start_trace(
 
                 client.end_trace(request_id)
         """
+        # Validate no active trace is set in the global context. If there is an active trace,
+        # the span created by this method will be a child span under the active trace rather than
+        # a root span of a new trace, which is not desired behavior.
+        if span := mlflow.get_current_active_span():
+            raise MlflowException(
+                f"Another trace is already set in the global context with ID {span.request_id}. "
+                "It appears that you have already started a trace using fluent APIs like "
+                "`@mlflow.trace()` or `with mlflow.start_span()`. However, it is not allowed "
+                "to call MlflowClient.start_trace() under an active trace created by fluent APIs "
+                "because it may lead to unexpected behavior. To resolve this issue, consider the "
+                "following options:\n"
+                " - If you want to create a child span under the active trace, use "
```

Comment:
```suggestion
                " - To create a child span under the active trace, use "
```

I'd remove `If you want`.